addons/ondevice-actions/src/components/ActionLogger/Inspect.tsx

```diff
@@ -20,123 +20,121 @@ const theme = {
   ARROW_ANIMATION_DURATION: '0',
 };
 
-class Inspect extends Component<{ name?: string; value: any }, { expanded: boolean }> {
-  state = { expanded: false };
+interface InspectProps {
+  name?: string;
+  value: any;
+}
 
-  render() {
-    const { name, value } = this.props;
-    const { expanded } = this.state;
-    const toggle = (
-      <View style={styles.container}>
-        {name &&
-        ((Array.isArray(value) && value.length) ||
-          (value &&
-            typeof value === 'object' &&
-            !Array.isArray(value) &&
-            Object.keys(value).length)) ? (
-          <Button
-            onPress={() => this.setState((s) => ({ expanded: !s.expanded }))}
-            title={!expanded ? '▶' : '▼'}
-          />
-        ) : null}
-      </View>
-    );
+const Inspect = ({ name, value }: InspectProps) => {
+  const [expanded, setExpanded] = useState(false);
+  const toggle = (
+    <View style={styles.container}>
+      {name &&
+      ((Array.isArray(value) && value.length) ||
+        (value &&
+          typeof value === 'object' &&
+          !Array.isArray(value) &&
+          Object.keys(value).length)) ? (
+        <Button
```

Comment:
Not necessary for this pr I guess but it might be good to replace this button with a custom touchable opacity or pressable to improve the UI for the Android version.

Reply:
Good idea! Didn't do that yet, but we could tackle it either in this PR or as a separate PR.